### ISSUE TITLE
Remove container_name from docker-compose.yml

### ISSUE THREAD

**Issue by ashsmith** [magento/magento2devbox-web#34]
I don't understand why we need this, I'd like to understand why?

I noticed the bash scripts depend on this, but they can be changed to use `docker-compose` commands. See below for an example:

    # Before:
    web_port=$(docker port magento2devbox_web_2134684c495edc8adf4b3eef983e61e7 80)
    # After:
    web_port=$(docker-compose port web 80)

    # Before:
    varnish_port=$(docker port magento2devbox_varnish_2134684c495edc8adf4b3eef983e61e7 6081)
    # After:
    varnish_port=$(docker-compose port varnish 6081)

**Comment by vrann** [comment]
@ashsmith Container name is needed in order to make it possible to start multiple projects at the same time. The use case is next: you are working on two completely different codebases in parallel and would like to edit and debug them at the same time.

Solution: you start two different docker compose clusters for each project. Same services in the clusters will have unique name and ports across the system. It will stay unique after restarting of the cluster.

Docker by default does not allow this functionality and usually uses the directory name as a container name, which is not unique and leads to collisions. 

Another solution would be to have same cluster but different "web" containers for different projects. It is possible but is not manageable through Web UI tool.

Let me know if you see how it can be done better.

**Comment by ashsmith** [comment]
@vrann True, that will become an issue. The way we get around this issue by using environment variables.

docker-compose will look for a `.env` file where you can change certain settings, including the project name

Take the following example `.env` file from a client project:

    COMPOSE_PROJECT_NAME=myawesomeclient
    COMPOSE_FILE=docker-compose-common.yml:docker-compose-dev.yml

Perhaps this is a nicer solution? Rather than having to write the unique name to the docker-compose.yml file.

**Comment by vrann** [comment]
@ashsmith that's great suggestion if I can get it working

Tried this:
- created project.env file in the build-test-env directory (where my docker-compose.yml and the rest of the scripts)
- added the env variable there: COMPOSE_PROJECT_NAME=1ef9e71c88dda3647e1ec60947ff5599
- run docker-compose up -d

Expected:
```
Creating 1ef9e71c88dda3647e1ec60947ff5599_db_1
Creating 1ef9e71c88dda3647e1ec60947ff5599_web_1
Creating 1ef9e71c88dda3647e1ec60947ff5599_varnish_1
```
Actual:
```
Creating buildtestenv_db_1
Creating buildtestenv_web_1
Creating buildtestenv_varnish_1
```

**Comment by ashsmith** [comment]
@vrann It has to be named `.env`, `project.env` won't work unfortunately. Docs over here: https://docs.docker.com/compose/env-file/

**Comment by vrann** [comment]
@ashsmith my bad, I read it like *.env

Works perfectly! Will add the suggestion to the backlog
